### PR TITLE
Issue #3017760: Support drush 9

### PR DESCRIPTION
**Pull request by malcomio**
work in progress...

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -7,7 +7,11 @@
     "require": {
         "edisonlabs/merge-yaml": "~1"
     },
-    "require-dev": {
-        "drush/drush": "~8.1.17"
```
Comment (manuel-appnovation) [inline composer.json]: We currently need this for the tests.

Comment (manuee) [inline composer.json]: We currently need this for the tests.

**Review by vijaycs85 — APPROVED** [review]
Looks good to me!

File: src/Commands/DatabaseSanitizeCommands.php [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+      ],
+    ];
+    foreach ($missing_tables as $table) {
+      $content['sanitize'][$machine_name][$table] = [
+        'description' => "Sanitization entry for {$table}. Generated by drush db-sanitize-generate.",
+        'query' => "TRUNCATE TABLE {$table}",
+      ];
+    }
+
+    return $content;
```
Comment (malcomio) [inline src/Commands/DatabaseSanitizeCommands.php:112]: still need to do something with this return value, not quite sure what

trying to find an example where drush writes to YML files

Comment (manuee) [reply src/Commands/DatabaseSanitizeCommands.php:112]: We do not need to write directly to a file. This is fine as it is.

If the user wants to write the results to a file, all they have to do is direct the output to one, like so:
`drush dbsg --machine-name=my_profile > myfile.yml`

File: src/Commands/DatabaseSanitizeCommands.php [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+/**
+ * A Drush commandfile.
+ *
+ * In addition to this file, you need a drush.services.yml
+ * in root of your module, and a composer.json file that provides the name
+ * of the services file to use.
+ *
+ * See these files for an example of injecting Drupal services:
+ *   - http://cgit.drupalcode.org/devel/tree/src/Commands/DevelCommands.php
+ *   - http://cgit.drupalcode.org/devel/tree/drush.services.yml
```
Comment (manuee) [inline src/Commands/DatabaseSanitizeCommands.php:16]: This docblock needs updating to describe the class itself.

File: src/Commands/DatabaseSanitizeCommands.php [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+    if (!empty($options['list'])) {
+      $this->logger()->warning(implode("\n", $missing_tables));
+    }
+  }
+
+  /**
+   * Generates a database.sanitize.yml file.
+   *
+   * Generate database.sanitize.yml file for tables not specified on sanitize
+   * YML files.
```
Comment (manuee) [inline src/Commands/DatabaseSanitizeCommands.php:64]: This isn't correct, we do not generate a `database.sanitize.yml` file here, we just generate the content for it.

**Review by manuee — CHANGES_REQUESTED** [review]

File: drush.services.yml [hunk 5]
```diff
@@ -0,0 +1,5 @@
+services:
+  database_sanitize.commands:
+    class: \Drupal\database_sanitize\Commands\DatabaseSanitizeCommands
+    tags:
```
Comment (manuee) [inline drush.services.yml:4]: Lets inject here the `database_sanitize` service:
`arguments: ['@database_sanitize']`

Then add a constructor like on http://cgit.drupalcode.org/devel/tree/src/Commands/DevelCommands.php

**Review by manuee — CHANGES_REQUESTED** [review]

File: drush.services.yml (hunk 5 above)
Comment (malcomio) [reply drush.services.yml:4]: good idea - I probably won't get time to work on this over the next few days, so if you have time please could you take this forward?

**Comment by manuee** [comment]
Lets continue the work on the new branch 3017760

**Comment by manuee** [comment]
https://github.com/EdisonLabs/database_sanitize/pull/13

**Pull request merged**